std/algorithm/iteration.d

```diff
@@ -2917,6 +2918,14 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR)
     {
         private RoR _items;
         private ElementType!RoR _current;
+        bool inputStartsWithEmpty = false;
+        static if (isBidirectional)
+        {
+            private ElementType!RoR _currentBack;
+            bool inputEndsWithEmpty = false;
+        }
+        enum isBidirectional = isBidirectionalRange!RoR &&
+                               isBidirectionalRange!(ElementType!RoR);
```

Comment:
Shouldn't this be `isBidirectionalRange!Separator` ? Why are the elements of `RoR` supposed to be bidirectional ranges?